src/classes/TDTM_TriggerHandler.cls

```diff
@@ -181,15 +182,19 @@ public with sharing class TDTM_TriggerHandler {
     
     private void runAsync(TDTM_Runnable classToRun, String classToRunName, List<Sobject> newList, 
     List<Sobject> oldList, TDTM_Runnable.Action thisAction, Schema.DescribeSobjectResult describeObj) {
-       Map<Id,Sobject> nm;
-       if(newlist != null) nm = new Map<Id,Sobject>(newlist);
-       else nm = new Map<Id,Sobject>();
+       set<Id> setNewId;
```

Comment:
the point is to actually pass null if newlist or oldlist are not provided by the trigger.  your code would end up passing an empty list to the trigger handler, and several would hit errors due to assuming that if oldlist was provided, that its size matched newlist's size.  the semantics of triggers are that trigger.old or trigger.new are null if they are not provided.